src/frontend/js/components/DashboardCreditCardsManagement/DashboardEditCreditCard.tsx

```diff
@@ -0,0 +1,186 @@
+    defaultMessage: 'Delete',
+  },
+  titleInputLabel: {
+    id: 'components.DashboardEditCreditCard.titleInputLabel',
+    description: 'Label of the title input on the credit card edit page',
+    defaultMessage: 'Name of the credit card',
+  },
+  lastNumbersInputLabel: {
+    id: 'components.DashboardEditCreditCard.lastNumbersInputLabel',
+    description: 'Label of the last numbers input on the credit card edit page',
+    defaultMessage: 'Numbers',
+  },
+  expirationInputLabel: {
+    id: 'components.DashboardEditCreditCard.expirationInputLabel',
+    description: 'Label of the expiration input on the credit card edit page',
+    defaultMessage: 'Expiration',
+  },
+  isMainInputLabel: {
+    id: 'components.DashboardEditCreditCard.isMainInputLabel',
+    description: 'Label of the "is_main" input on the credit card edit page',
+    defaultMessage: 'Use this credit card as default',
+  },
+});
+
+interface Props {
+  creditCard: CreditCard;
+  onSettled?: Function;
+}
+
+interface FormValues {
+  title: string;
+  last_numbers: string;
+  expiration: string;
+  is_main: boolean;
+}
+
+export const DashboardEditCreditCard = ({ creditCard, onSettled }: Props) => {
+  const intl = useIntl();
+
+  const {
+    methods: { update, safeDelete },
+    states: { isLoading, error },
+  } = useCreditCardsManagement();
+
+  const defaultValues: FormValues = {
+    title: '',
+    last_numbers: '',
+    expiration: '',
+    is_main: true,
+  };
+
+  const { register, handleSubmit, reset, formState } = useForm<FormValues>({
+    defaultValues,
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
+
+  useEffect(() => {
+    reset({
+      ...creditCard,
+      expiration: creditCard.expiration_month + '/' + creditCard.expiration_year,
+      last_numbers: '•••• ••••  ••••  ' + creditCard.last_numbers,
+    });
+  }, [creditCard]);
+
+  const onSubmit = (data: FormValues) => {
+    update(
+      {
+        ...creditCard,
+        title: data.title,
+        is_main: data.is_main,
+      },
+      {
+        onSuccess: () => {
+          onSettled?.();
+        },
+      },
+    );
+  };
+
+  const renderContent = () => {
+    if (isLoading) {
+      return <Spinner />;
+    }
+    if (error) {
+      return <Banner message={error} type={BannerType.ERROR} rounded />;
+    }
+    return (
+      <form>
+        <TextField
+          aria-invalid={!!formState.errors.title}
+          id="title"
+          label={intl.formatMessage(messages.titleInputLabel)}
+          error={!!formState.errors.title}
+          message={getLocalizedErrorMessage(intl, formState.errors.title?.message)}
+          {...register('title')}
+        />
+        <div className="dashboard-edit-credit-card__sensitive">
+          <TextField
+            id="last_numbers"
+            disabled={true}
+            label={intl.formatMessage(messages.lastNumbersInputLabel)}
+            {...register('last_numbers')}
```

Comment:
No need to register disabled field. There are only display in an informative purpose.
```suggestion
```

Reply:
It is mandatory otherwise the field value will not be set and we will have an empty field.